### PR TITLE
k8s: handle ``fail_json`` API from inventory plugin

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* inventory plugin does not have module object so fail_json
  will not work. Use AnsibleError instead of ``fail_json``

Fixes: #57

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/fail_json.yml
plugins/inventory/k8s.py
plugins/module_utils/common.py
plugins/module_utils/exceptions.py

### CODE REVIEW

**Comment by Akasurde** [comment]
recheck

**Review by abikouo — APPROVED** [review]
LGTM

File: plugins/module_utils/common.py
Comment (alinabuzachis) [inline plugins/module_utils/common.py:34]: Should it be from `kubernetes.core` instead of `community.kubernetes`?

Comment (Akasurde) [reply plugins/module_utils/common.py]: Yes. Thanks for correcting me.

File: plugins/inventory/k8s.py
Comment (alinabuzachis) [inline plugins/inventory/k8s.py:121]: @Akasurde even here I guess.

**Review by gravesm — APPROVED** [review]
We still need to backport the community.kubernetes -> kubernetes.core changes to this branch, so for now it should be fine to remain as community.kubernetes. As soon as this gets merged I'll do that backport.

**Review by alinabuzachis — APPROVED** [review]

**Review by gravesm — APPROVED** [review]

**Comment by JiffsMaverick** [comment]
Today is July. There were still no fixes in versions 1.x or 2.x. @gravesm could you backport these fixes to main branch?

**Comment by gravesm** [comment]
@JiffsMaverick Are you seeing an error? If so, could you create a new issue with the collection version and steps to reproduce?

**Comment by JiffsMaverick** [comment]
@gravesm Sorry, I didn't check it carefully. I checked again. There is a fix in both 1.x and 2.x. BUT! The problem is that version 1.2.2 (or 1.3.0?) hasn't been released yet. And version 2.1.1 is not included in Ansible 3/4 yet. Ok, I'll wait for the new releases...

**Pull request merged**
